modules/serverless-functions-func-yaml-fields.adoc

```diff
@@ -11,7 +11,12 @@ Many of the fields in `func.yaml` are generated automatically when you create, b
 [id="serverless-functions-func-yaml-builder_{context}"]
 == builder
 
-The `builder` field specifies the Buildpack builder image to use when building the function. In most cases, this value should not be changed. When you do change it, use a value that is listed in the `builders` field.
+The `builder` field specifies the builder image to use when building the function. The accepted values for this field are `S2i` and `pack`. Using the `S2i` value means that the function uses the link:https://cloud.redhat.com/blog/create-s2i-builder-image[source-to-image (S2I)] image builder. Using `pack` means that the builder uses the link:https://buildpacks.io/[Buildpacks] image builder.
```

Comment:
@abrennan89
I am afraid this is not accurate.
The `builder` field of `func.yaml` is container image, e.g. `gcr.io/paketo-buildpacks/builder:base` not `s2i` or `pack`.

However the `--builder` flag was repurposed: previously it was container image, now it is build-type i.e. `s2i` or `pack`.

In addition there is a new flag `--builder-image` that has same meaning as the `--builder` had before.

Am I right @lkingland ?

Reply:
I hope I am wrong.

Reply:
It seems very confusing to me that we use the same words for so many things that mean something different.
@matejvasek is there a link for upstream docs for this somewhere? I couldn't find any when I looked.

I will probably just close this PR and open a new one since this one isn't accurate, but I think we need to clarify the docs here around what is in func.yaml vs what is used by CLI and the relationship between them. It's not a good UX to use different words for the same things and the same words for things that mean different things.

Reply:
> However the CLI --builder flag was repurposed: previously it was container image, now it is build-type i.e. s2i or pack.

In that case I think in the YAML it should be renamed from builder > container-image to make this less confusing.

Reply:
>In addition there is a new flag --builder-image that has same meaning as the --builder had before.

Will this be the new flag going forward? From a user / UX perspective I think that builder-image is a very confusing name, builder makes more sense IMHO, since builder-image feels like it's confusing build types and container image together

Reply:
I agree, probably we should keep meaning of `--builder` and rather and new flag named `--builder-type`. wdyt @lkinglang ? I personally don't like flag re-purposing -- it invalidates my shell history :disappointed:

Reply:
Or we could align it other way around change `builder` -> `builder-image` in func.yaml. Definitely the flag and the field should be named the same if possible.

Reply:
https://github.com/knative-sandbox/kn-plugin-func/issues/942

Reply:
This documentation is correct when discussing the _flag_ `--builder`.  This value will most likely be removed from func.yaml

Reply:
OK, I will revisit this in a new PR maybe and just leave the func.yaml alone. Is it worth trying to document this now if the plan is to change it to `--strategy`? It seems like documenting the old version and then updating it will cause unnecessary confusion rather than just introducing it once later when the naming is properly decided?